infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -359,8 +355,7 @@ private void applyShadowStream(UUID streamId, long snapshot) {
             if (bufferSize + smrEntry.getSerializedSize() >
                 logReplicationMetadataManager.getRuntime().getParameters()
                     .getMaxWriteSize()) {
-                try (TxnContext txnContext =
-                    logReplicationMetadataManager.getTxnContext()) {
+                try (TxnContext txnContext = logReplicationMetadataManager.getTxnContext()) {
```

Comment:
If a given stream is not in the registry table, or has is_federated flag false, it should not be applied.  I did not seem to find where it is done.